File: JavaScript/20-有效括号匹配.js
var isValid = function(s) {
    if(s == null||s.length <1){
        return true;
    }
    var stack = [];
    for(var i=0;i<s.length;i++){
        if(s[i] == ""){
            continue;
        }
        if(stack.leng<1){
            stack.push(s[i]);
        }else if(stack[stack.length-1]=="("&&s[i]==")" ||stack[stack.length-1]=="{"&&s[i]=="}" ||stack[stack.length-1]=="["&&s[i]=="]"){
            stack.pop();
        }else{
            stack.push(s[i]);
        }
    }
    return stack.length<1;
};

File: LeetCode-Go/20-有效括号匹配.go
package main

func isValid(s string) bool {
	stack := make([]byte, 0)
	for _, v := range s {
		if v == '[' || v == '(' || v == '{' {
			stack = append(stack, byte(v))
		} else {
			if (len(stack) != 0 && match(stack[len(stack)-1], byte(v))) {
				stack = pop(stack)
				continue
			} else {
				return false
			}
		}
	}

	if (len(stack) == 0) {
		return true
	} else {
		return false
	}
}
func match(left, right byte) bool {
	switch right {
	case ']':
		if left != '[' {
			return false
		}
	case ')':
		if left != '(' {
			return false
		}
	case '}':
		if left != '{' {
			return false
		}
	}
	return true
}

func pop(stack []byte) []byte {
	return stack[:len(stack)-1]
}
